tests/Microsoft.Identity.Test.Unit/PublicApiTests/ObjectsAndInterfaces.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.PublicApiTests
+{
+    [TestClass]
+    public class ObjectsAndInterfaces
+    {
+        [TestMethod]
+        [Description(
+            "This test checks that we don't accidently expose a public property or method on" +
+            " an object and forget to expose it on the associated interface.")]
+        public void InterfaceExposesPublicSurfaceOfObjects()
```

Comment:
This should protect against future similar bugs where we expose a public property / method on the obj but not on the interface.